ragna/source_storages/_embedding_model.py

```diff
@@ -0,0 +1,47 @@
+from abc import ABC, abstractmethod
+
+from sentence_transformers import SentenceTransformer
+import torch
+
+from ragna.core._components import Component
+
+from ragna.source_storages._vector_database import Chunk
+
+device = 'cuda' if torch.cuda.is_available() else 'cpu'
```

Comment:
1. We cannot hardcode this. What if the user wants to run the embedding on their CPU even if they have a GPU? This being a default is fine though.
2. What is the performance when running this model on CPU? How many tokens per second can we expect here? I'd rather not include a model that basically requires a GPU to be somewhat usable.